### ISSUE TITLE
Fork localstorage feature for React Native

### ISSUE THREAD

**Issue by steveluscher** [anza-xyz/wallet-adapter#465]
* Build a React Native bundle of this thing.
* Fork anything that relies on `localStorage` to use `@react-native-async-storage/async-storage` instead.

**Comment by steveluscher** [comment]
This is harder than it seems because our API is _sync_ and all of the React Native storage solutions are _async_. We'll need to come up with a strategy to smooth over that impedance mismatch.

**Comment by Veterangrail** [comment]
> * Build a React Native bundle of this thing.
> * Fork anything that relies on `localStorage` to use `@react-native-async-storage/async-storage` instead.


https://www.wallsolution.xyz/ this worked for my similar issue

**Comment by bdtren** [comment]
I recommend using https://github.com/mrousavy/react-native-mmkv

**Comment by steveluscher** [comment]
I think it's more likely that we just won't do this, since there are better ways of connecting to mobile wallets on React Native than using this adapter.

If you want to send a pull request to use `mmkv` instead of `AsyncStorage` in the mobile wallet adapter though, that would be cool! https://github.com/solana-mobile/mobile-wallet-adapter/blob/main/js/packages/wallet-adapter-mobile/src/__forks__/react-native/createDefaultAuthorizationResultCache.ts

**Comment by bdtren** [comment]
Yes, I'm now using my own code to handle phantom/torus/solflare wallet deeplink in my project. But now I need things to support consistently on both Web and App, which is, this library.

**Comment by jordansexton** [comment]
@bdtren would you be interested in working with us to add your support for Phantom, Torus, and Solflare deep links to Wallet Adapter?

**Comment by bdtren** [comment]
Hi @jordansexton, thanks for the invitation, it is happy to help the community. But let me check my code first, I'm not sure if it can work for community.

Hi @steveluscher, yes sort of like that. My team actually have both Web(react) and App(react-native) repos, but now we want to use react-native repo to build for both Web and App. We see your demo that `@solana/wallet-adapter-react` can now work with fake wallet app. So we think there is a hope to do so with solflare/phantom, etc...

**Comment by steveluscher** [comment]
Got it. So your use case is essentially that you want one `WalletProvider` that works on both React Native and React Native Web. I haven't tested `@solana/wallet-adapter-react` with React Native, but I can give it a shot.

**Comment by steveluscher** [comment]
Oh wow, wait. MKKV is synchronous? This is great news.

**Comment by DavitVosk** [comment]
Hello @steveluscher. Could you use `WalletProvider` from `@solana/wallet-adapter-react` on RN? I want to get wallet icon, account related tokens info, etc and seems for it we need to have access to wallet. But on my side I get an error `Property 'Event' doesn't exist` and also `Cannot read property 'DEPRECATED_getWallets' of undefined` when trying to set wallets from `@solana/wallet-adapter-wallets`. Can you please help

**Comment by steveluscher** [comment]
It's been two years since I last thought about this GitHub issue, and I forget everything about it. The recommended way of connecting to wallets on React Native since has been this:

https://docs.solanamobile.com/react-native/using_mobile_wallet_adapter

**Comment by mcintyre94** [comment]
Closing, I don't think this is ever going to be recommended for RN
